suite-native/module-passphrase/src/screens/PassphraseLoadingScreen.tsx

```diff
@@ -1,37 +1,56 @@
 import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
 
 import { useNavigation } from '@react-navigation/native';
 
 import { VStack, Loader } from '@suite-native/atoms';
 import {
     AppTabsRoutes,
     HomeStackRoutes,
+    PassphraseStackParamList,
+    PassphraseStackRoutes,
     RootStackParamList,
     RootStackRoutes,
     Screen,
-    StackNavigationProps,
+    StackToStackCompositeNavigationProps,
 } from '@suite-native/navigation';
+import {
+    selectIsDeviceAccountless,
+    selectIsDeviceDiscoveryActive,
+} from '@suite-common/wallet-core';
 
 import { PassphraseScreenHeader } from '../components/PassphraseScreenHeader';
 
-type NavigationProp = StackNavigationProps<RootStackParamList, RootStackRoutes.PassphraseStack>;
+type NavigationProp = StackToStackCompositeNavigationProps<
+    PassphraseStackParamList,
+    PassphraseStackRoutes,
+    RootStackParamList
+>;
 
 export const PassphraseLoadingScreen = () => {
     const navigation = useNavigation<NavigationProp>();
+    const isDeviceAccountless = useSelector(selectIsDeviceAccountless);
+    const isDiscoveryActive = useSelector(selectIsDeviceDiscoveryActive);
 
     useEffect(() => {
-        // NOTE: This is just for demo purposes. Proper loading screen will be implemented in a follow-up.
-        const timeout = setTimeout(() => {
-            navigation.navigate(RootStackRoutes.AppTabs, {
-                screen: AppTabsRoutes.HomeStack,
-                params: {
-                    screen: HomeStackRoutes.Home,
-                },
-            });
+        setTimeout(() => {
```

Comment:
why was the `clearTimeout` removed?

Reply:
https://github.com/trezor/trezor-suite/pull/12193/commits/fd61a43929e4a65804e39a40a5c1d039432da8cf